modules/aws-kubernetes/acm-certificate/main.tf

```diff
@@ -0,0 +1,49 @@
+## acm certificate request
+resource "aws_acm_certificate" "main" {
+  domain_name               = "${var.domain_name}"
+  subject_alternative_names = "${var.sans}"
+  validation_method         = "DNS"
+
+  lifecycle  {
+    create_before_destroy = true
+  }
+}
+
+## acm certificate records for validation
+resource "aws_route53_record" "main-0" {
```

Comment:
Why do you create `main-0` and `main-1` aws_rourte53_record? Can it be achieve with a `count` attribute and a single resource `main`?

Reply:
yep, you're right! this is a stylish refactoring (i will change that asap) even first as i asked, i would like what do you think about the idea of changing the behaviour mainly

Reply:
i am always of the idea of 

1. make it works
1. make it fast
1. make it better

now the pr is even before the point 1 afaik, that is "could be an acceptable implementation?"

Reply:
Sure! You're right.

For sure, all movements to be more "automated" it's welcomed so changing the current behavior (having a certificate already available in acm) to one more automated (generating a certificate) it's nice.

Consider to make "migration" tests (after making it working). Thanks for your work!